discord/commands/context.py

```diff
@@ -70,6 +79,39 @@ def __init__(self, bot: "discord.Bot", interaction: Interaction):
         self.command: ApplicationCommand = None  # type: ignore
         self._state: ConnectionState = self.interaction._state
 
+    async def invoke(self, command: ApplicationCommand, /, *args: P.args, **kwargs: P.kwargs) -> T:
```

Comment:
Using `ParamSpec` here doesn't work because it's only used once. An example of ParamSpec being used is in `commands.Context.invoke`:

https://github.com/Pycord-Development/pycord/blob/4d8b3d939cfda3792af0ea51188e9835dd6c1905/discord/ext/commands/context.py#L157

So you'd most likely need to make `ApplicationCommand` generic for this to work.